crates/fuzzing/src/generators/table_ops.rs

```diff
@@ -43,32 +46,61 @@ impl TableOps {
     /// The "run" function is guaranteed to terminate (no loops or recursive
     /// calls), but is not guaranteed to avoid traps (might access out-of-bounds
     /// of the table).
-    pub fn to_wat_string(&self) -> String {
-        let mut wat = "(module\n".to_string();
+    pub fn to_wasm_binary(&self) -> Vec<u8> {
+        let mut module = Module::new();
 
         // Import the GC function.
-        wat.push_str("  (import \"\" \"gc\" (func))\n");
+        let mut imports = ImportSection::new();
+        imports.import("", "gc", ImportType::Function(0));
 
         // Define our table.
-        wat.push_str("  (table $table ");
-        write!(&mut wat, "{}", self.table_size()).unwrap();
-        wat.push_str(" externref)\n");
+        let mut tables = TableSection::new();
+        tables.table(TableType {
+            element_type: ValType::ExternRef,
+            limits: Limits {
+                min: self.table_size(),
+                max: None,
+            },
+        });
+
+        // Encode the type section for the "run" function.
+        let mut types = TypeSection::new();
+        let mut params: Vec<ValType> = Vec::with_capacity(self.num_params() as usize);
+        for _i in 0..self.num_params() {
+            params.push(ValType::ExternRef);
+        }
+        let results = vec![];
+        types.function(params, results);
 
         // Define the "run" function export.
-        wat.push_str(r#"  (func (export "run") (param"#);
-        for _ in 0..self.num_params() {
-            wat.push_str(" externref");
+        let mut functions = FunctionSection::new();
+        functions.function(0);
+
+        let mut exports = ExportSection::new();
+        exports.export("run", Export::Function(0));
```

Comment:
The `0`th function is the gc function import (the function index space goes `[import function 0, import function 1, import function 2, ..., defined function 0, defined function 1, defined function 2, ...]`) so this index should be `1` here.